task3/middleware.py

```diff
@@ -0,0 +1,17 @@
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+class CustomAuthMiddleware:
+    def __init__(self, get_response):
+        self.get_response = get_response
+
+    def __call__(self, request):
+        if not request.user.is_authenticated:
```

Comment:
After checking if user is not `authenticated` redirect it to login or authenticate it with custom authentication. In `else` part too if user is authenticated redirect it to some link or home page

Reply:
ok